src/NServiceBus.Hosting.AcceptanceTests/When_starting_and_stopping_host.cs

```diff
@@ -0,0 +1,78 @@
+﻿namespace NServiceBus.AcceptanceTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class When_starting_and_stopping_host
+    {
+        [Test]
+        public Task Start_and_stop_are_called()
+        {
+            var defaultProfiles = new List<Type>() { typeof(Lite) };
+            string[] args = { };
+            var context = new IWantToRunContext();
+            IConfigureThisEndpoint configurer = new GenericEndpointConfig(context);
+
+            var host = new GenericHost(configurer, args, defaultProfiles, GenericEndpointConfig.EndpointName);
+
+            host.Start();
+            host.Stop();
+
+            Assert.True(context.StartCalled);
+            Assert.True(context.StopCalled);
+
+            return Task.FromResult(0);
+        }
+
+        class IWantToRunContext
+        {
+            public bool StartCalled { get; set; }
+            public bool StopCalled { get; set; }
+        }
+
+        class GenericEndpointConfig : IConfigureThisEndpoint
+        {
+            IWantToRunContext context;
```

Comment:
Should be at the bottom of the file  (run a R# code cleanup with the "particular" conventions)